### ISSUE TITLE
gtags error! when I tried to update database manually as it can't update automatically when I switch git branch.

### ISSUE THREAD

**Issue by JaminYoung** [Yggdroot/LeaderF#1040]
- vim or neovim? 
    - [x] vim
    - [ ] neovim
- Output of `vim --version` or `nvim --version`: 8.2
- Output of `:echo has("python")`: 0
- Output of `:echo has("python3")`: 1
- Output of `:echo &pythondll`(only vim, not neovim): 
- Output of `:echo &pythonthreedll`(only vim, not neovim): 
- Output of `:py print(sys.version)`: 
- Output of `:py3 print(sys.version)`: 3.10.12
- Output of `:echo g:Lf_Debug_Cmd`: 
- Output of `:echo g:Lf_FilesFromCache`: 
- Operating system: 
    - [x] Linux(WSL)
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Configurations related to LeaderF in vimrc: 
![image](https://github.com/Yggdroot/LeaderF/assets/25146444/4c8802f4-7c04-4406-9e89-2571f352562b)

### Describe your question, feature request, or bug.
I switch to another git branch, but gtags database didn't update automatically, so I tried to update manually. The first several times were susccesful, but then went wrong, showing gtags error, type `:mess` showed bellow. 
![image](https://github.com/Yggdroot/LeaderF/assets/25146444/4a95f030-99b6-40f6-bbfc-ca0974dc8c24)
I deleted the gtags database, and tried the same procedure, and it happened again.

### Steps to reproduce
0.` gtags --version` 6.6.10
1. cd to a repository,（ I git cloned uboot repository)
2. start vim, auto generate database successfully
3. exit vim
4. git checkout to another branch
5. start vim
6. type `:Leaderf gtags --update` 
(first time maybe success, by switch back former branch, try update gtags database again ,it will go wrong)

### Actual behaviour
`Leaderf gtags --update` showing gtags error!

### Expected behaviour
gatags database should update automatically after switch to another branch, and should not go wrong by update manually?

**Comment by JaminYoung** [comment]
update:
I found out each time I tried to update to gtags database by `:Leaderf gtags --update`, the size of gtags database will increased abnormaly, see below
![image](https://github.com/Yggdroot/LeaderF/assets/25146444/d05035a3-c3d3-4178-9f30-811a358175a2)
and `:mess`, first time went error actually showed `malloc(): corrupted top size`, 
![image](https://github.com/Yggdroot/LeaderF/assets/25146444/e41a9c70-60c8-41e1-9019-cfced76a383e)

 And I could reproduce it by just type
`gtags -i --skip-symlink --gtagslabel native-pygments` in terminal.
But by removing the param `-i` will runs fine. I looked up the help file of gtags, `-i` stands for `increased`, still not know why.
